src/nvim/screen.c

```diff
@@ -6888,6 +6889,30 @@ static void draw_tabline(void)
   int use_sep_chars = (t_colors < 8
                        );
 
+  if (ui_is_widget_external(kUITabline)) {
```

Comment:
Let's factor this out to a new function, `draw_tabline_ext()`.

```c
if (ui_is_widget_external(kUITabline)) {
  draw_tabline_ext();
  return;
}
```

If possible, try to follow a similar pattern for other externalized widgets.